Model/Source/JobMethods.php

```diff
@@ -7,14 +7,18 @@ class JobMethods implements \Magento\Framework\Data\OptionSourceInterface
     private $methods = [
         'saveConfigurationToAlgolia' => 'Save Configuration',
         'moveIndexWithSetSettings' => 'Move Index',
-        'deleteObjects' => 'Object deletion',
-        'rebuildStoreCategoryIndex' => 'Category Reindex',
-        'rebuildCategoryIndex' => 'Category Reindex',
-        'rebuildStoreProductIndex' => 'Product Reindex',
-        'rebuildProductIndex' => 'Product Reindex',
-        'rebuildStoreAdditionalSectionsIndex' => 'Additional Section Reindex',
-        'rebuildStoreSuggestionIndex' => 'Suggestion Reindex',
-        'rebuildStorePageIndex' => 'Page Reindex',
+        'rebuildIndex' => 'Rebuild Full Index',
+        'rebuildIndexIds' => 'Rebuild Specific IDs',
+        'deleteInactiveProducts' => 'Delete Inactive Products',
+        // @deprecated
+        'deleteObjects' => 'Object deletion (deprecated)',
+        'rebuildStoreCategoryIndex' => 'Category Reindex (deprecated)',
```

Comment:
I like this approach. This looks like a graceful way to handle this migration to the new methods. Have we tested mixed queues post migration from a previous version (< `3.15`)? Any potential issues you can think of with that?

Reply:
Yes ! I filled the queue with a full reindexing in the queue **before** and **after** applying those changes and the queue is processed as it should be (obviously the new jobs can't be merged with the old one because the process checks for the class and the method name to merge the jobs, but I think this is not a big deal)